lib/racecar/config.rb

```diff
@@ -103,6 +103,9 @@ class Config < KingKonf::Config
     desc "The SCRAM mechanism to use, either `sha256` or `sha512`"
     string :sasl_scram_mechanism, allowed_values: ["sha256", "sha512"]
 
+    desc "The OAUTHBEARER token provider class"
+    string :sasl_oauth_token_provider
```

Comment:
so type here is `string` but you're sending a `class` did you test if this is working? could you also provide an example?

Reply:
Yes, it does work!

```
Racecar.configure do |config|
  config.brokers = "localhost:9292"
  config.client_id =  "client"
  config.sasl_oauth_token_provider =  KafkaTokenProvider.new
  config.ssl_ca_certs_from_system = true
end
```

Reply:
The config is based on [KingKonf](https://github.com/dasch/king_konf) which is designed around the idea of pure data configuration values overridable via env vars or YAML. A token class isn't exactly data, so it might be best to confine the possibility to define it to just runtime (not a huge fan of poking holes through abstractions like specifying a class name, but maybe that's just me). Say, by making it just an `attr_accessor` on the config class.
It would also lose `desc`ription that way... but I don't see a better yet still simple way around this yet.

Reply:
Does this work when setting the config in an ENV var or the config file?

Reply:
env var only, since it required a class instance on it. Should I add this to the readme?

Reply:
I agree with @D-side-BL on this – it would be preferable to use an accessor on the Config class instead, since this setting doesn't follow the normal rules for KingKonf config.

Reply:
Either that, or using e.g. `get_const` to find the class based on a string config setting.

Reply:
ok I will change to `accessor` and I will add to the readme that this option only works for ruby config file.

Reply:
@dasch  can you check again please?